nl_server/config.py

```diff
@@ -40,102 +35,124 @@
+# object.
 #
-def load(embeddings_map: Dict[str, Dict[str, str]]) -> List[EmbeddingsIndex]:
-  # Create Index objects.
-  indexes = parse(embeddings_map)
-
-  #
-  # Download all the models.
-  #
-  models_set = set()
-  for i in indexes:
-    if i.model_name and i.model_type == ModelType.LOCAL:
-      models_set.add(i.model_name)
-  model2path = {d: gcs.download_folder(d) for d in models_set}
-  for idx in indexes:
-    if idx.model_name:
-      # the idx will only have a local path if it is a LOCAL model
-      idx.model_local_path = model2path.get(idx.model_name, "")
-
-  return indexes
+def parse(embeddings_map: Dict[str, Dict[str, str]]) -> EmbeddingsInfo:
+  if embeddings_map['version'] == 1:
+    return parse_v1(embeddings_map)
+  else:
+    logging.warning('Could not parse embeddings map: unsupported version.')
+    return None
 
 
 #
-# Parse the input `embeddings.yaml` dict representation, with structure:
-#   index-type => field (e.g., model) => value
+# Parses the v1 version of the `embeddings.yaml` dict representation into
+# EmbeddingsInfo object.
 #
-def parse(embeddings_map: Dict[str, Dict[str, str]]) -> List[EmbeddingsIndex]:
-  indexes: List[EmbeddingsIndex] = []
-
-  for key, value_map in embeddings_map.items():
-    # Let this fail if the enum isn't valid.
-    store_type = StoreType(value_map['store'])
-    path = value_map['embeddings']
-    model_name = value_map['model']
-    model_type = value_map.get('model_type', ModelType.LOCAL)
-
-    if path.startswith('/'):
-      # Value is an absolute path
-      file_name = os.path.basename(path)
-      local_path = path
-    elif is_gcs_path(path):
-      logging.info('Downloading embeddings from GCS path: %s', path)
-      if store_type == StoreType.MEMORY:
-        local_path = download_gcs_file(
-            path, use_anonymous_client=use_anonymous_gcs_client())
-      elif store_type == StoreType.LANCEDB:
-        local_path = gcs.download_folder(path)
-      if not local_path:
-        logging.warning(
-            'Embeddings not downloaded from GCS and will be ignored. Please check the path: %s',
-            path)
-        continue
-      file_name = path
+def parse_v1(embeddings_map: Dict[str, Dict[str, str]]) -> EmbeddingsInfo:
+  # parse the models
+  models = {}
+  for model_name, model_info in embeddings_map.get('models', {}).items():
+    model_type = model_info['type']
+    if model_type == ModelType.LOCAL:
+      models[model_name] = LocalModelInfo(type=model_type,
+                                          gcs_folder=model_info['gcs_folder'])
+    elif model_type == ModelType.VERTEXAI:
+      models[model_name] = VertexAiModelInfo(
+          type=model_type,
+          project_id=model_info['project_id'],
+          location=model_info['location'],
+          prediction_endpoint_id=model_info['prediction_endpoint_id'])
+    else:
+      logging.warning(
+          'Skip parsing information for model {model_name}: unsupported type {model_type}'
+      )
```

Comment:
should we be louder?

Reply:
logged an error instead, but could also raise an exception if you think this should be even louder? I just don't know if one bad model type should fail the whole loading of the embeddings, but I can also see an argument for the yaml having to be exactly as expected

Reply:
One bad model in embeddings.yaml is reason to fail this, especially if this code runs in tests, no? But of course, in custom DC, we should be more tolerant.

Reply:
done